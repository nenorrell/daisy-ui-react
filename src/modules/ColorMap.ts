import { Variant, VariantColors } from "../@types/Daisy";


interface ColorMap<K, V> extends Map<K, V> {
    get(key: K): V;
}

// Cast to ColorMap interface to get rid of inherit "possibly undefined value" from Map
export default <ColorMap<Variant, VariantColors>>new Map([
    ["primary", {
        bg: "bg-primary",
        text: "text-primary-content",
        focus: "primary-focus",
        btn: "btn-primary"
    }
    ],
    ["secondary", {
        bg: "bg-secondary",
        text: "text-secondary-content",
        focus: "secondary-focus",
        btn: "btn-secondary"
    }
    ],
    ["accent", {
        bg: "bg-accent",
        text: "text-accent-content",
        focus: "accent-focus",
        btn: "btn-accent"
    }
    ],
    ["neutral", {
        bg: "bg-neutral",
        text: "text-neutral-content",
        focus: "neutral-focus",
        btn: "btn-neutral"
    }
    ],
    ["info", {
        bg: "bg-info",
        text: "text-info-content",
        focus: "info-focus",
        btn: "btn-info"
    }
    ],
    ["success", {
        bg: "bg-success",
        text: "text-success-content",
        focus: "success-focus",
        btn: "btn-success"
    }
    ],
    ["error", {
        bg: "bg-error",
        text: "text-error-content",
        focus: "error-focus",
        btn: "btn-error"
    }
    ],
    ["warning", {
        bg: "bg-warning",
        text: "text-warning-content",
        focus: "warning-focus",
        btn: "btn-warning"
    }
    ],
    ["base-100", { bg: "bg-base-100", text: "text-base-content" }],
    ["base-200", { bg: "bg-base-200", text: "text-base-content" }],
    ["base-300", { bg: "bg-base-300", text: "text-base-content" }]
]);
