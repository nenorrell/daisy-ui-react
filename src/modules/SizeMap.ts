import { AllSizes, ComponentSize } from "../@types/Daisy";
import { IMap } from "../@types/Generic";

/**
 *
 * Why can't we just dynamically generate these class names instead of storing them like this?
 * See https://tailwindcss.com/docs/content-configuration#dynamic-class-names
*/
export default <IMap<AllSizes, ComponentSize>> new Map([
    ["xs", {
        btn: "btn-xs",
        badge: "badge-xs"
    }],
    ["sm", {
        btn: "btn-sm",
        badge: "badge-sm"
    }],
    ["md", {
        btn: "btn-md",
        badge: "badge-md"
    }],
    ["lg", {
        btn: "btn-lg",
        badge: "badge-lg"
    }],
    ["lg", {
        btn: "btn-lg",
        badge: "badge-lg",
    }],
    ["block", {
        btn: "btn-block"
    }],
    ["wide", {
        btn: "btn-wide"
    }]
]);

