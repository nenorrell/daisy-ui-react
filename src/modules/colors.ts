import { Variant } from "../@types/Colors";

export const getTextColor = (variant :Variant) =>{
    if(variant.includes("base")) {
        return "base";
    }
    return variant;
};
