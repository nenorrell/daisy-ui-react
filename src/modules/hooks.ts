import { useState, Dispatch, SetStateAction } from "react";

export function useToggle(initialValue: boolean = false): [boolean, Dispatch<SetStateAction<void>>] {
    const [value, setValue] = useState<boolean>(initialValue);
    const toggleValue = () => setValue(!value);
    return [value, toggleValue];
}
