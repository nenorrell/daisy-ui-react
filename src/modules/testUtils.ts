import { BaselessVariant, ButtonSize, Variant } from "../@types/Daisy";

export const BaselessVariantOptions :BaselessVariant[] = ["primary", "secondary", "accent", "neutral", "info", "success", "error", "warning"];
export const VariantOptions :Variant[] = [...BaselessVariantOptions, "base-100", "base-200", "base-300"];

export const ButtonSizes :ButtonSize[] = ["lg", "md", "sm", "xs"];
