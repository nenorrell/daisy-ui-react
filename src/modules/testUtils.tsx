import { BaselessVariant, StatusVariant, Variant } from "../@types/Colors";
import { ButtonSize, Size } from "../@types/Daisy";

export const BaselessVariantOptions :BaselessVariant[] = ["primary", "secondary", "accent", "neutral", "info", "success", "error", "warning"];
export const StatusVariantOptions :StatusVariant[] = ["info", "success", "error", "warning"];
export const VariantOptions :Variant[] = [...BaselessVariantOptions, ...StatusVariantOptions, "base-100", "base-200", "base-300"];

export const ComponentSizes :Size[] = ["lg", "md", "sm", "xs"];
export const ButtonSizes :ButtonSize[] = [...ComponentSizes, "block", "wide"];
