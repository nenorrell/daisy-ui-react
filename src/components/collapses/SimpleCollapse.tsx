import clsx from "clsx";
import { Variant } from "../../@types/Colors";
import ColorMap from "../../modules/ColorMap";
import { useToggle } from "../../modules/hooks";
import { Collapse } from "./Collapse";
import { ForwardedRef, forwardRef, PropsWithChildren, ReactNode } from "react";
import { CollapseTitle } from "./CollapseTitle";
import { CollapseBody } from "./CollapseBody";

interface ISimpleCollapse {
    headerContent :string
    LeftIcon ?:ReactNode
    className ?:string
    variant ?:Variant
    rounded ?:boolean
    defaultExpand ?:boolean
}
export const SimpleCollapse = forwardRef((
    {variant="neutral", ...props} :PropsWithChildren<ISimpleCollapse>,
    ref ?:ForwardedRef<HTMLDivElement>
)=>{
    const variantColors = ColorMap.get(variant);
    const [isExpanded, toggle] = useToggle(props.defaultExpand || false);

    return (
        <Collapse ref={ref} className={props.className}>
            <CollapseTitle
                onClick={toggle}
                className={clsx(
                    variantColors.bg,
                    variantColors.text,
                    isExpanded && "rounded-t-md",
                    !isExpanded && "rounded-md transition-all duration-[1s]"
                )}
            >
                { props.LeftIcon }
                <p className={clsx(
                    "w-full",
                    "text-lg",
                    props.LeftIcon && "ml-4"
                )}>{props.headerContent}</p>
            </CollapseTitle>

            <CollapseBody className={clsx(
                variantColors.bg,
                variantColors.text,
                props.rounded && "rounded-b-md"
            )}>
                {props.children}
            </CollapseBody>
        </Collapse>
    );
});

SimpleCollapse.displayName = "SimpleCollapse";
