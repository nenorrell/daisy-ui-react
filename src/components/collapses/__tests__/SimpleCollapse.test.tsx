import { faGithub } from "@fortawesome/free-brands-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { render } from "@testing-library/react";
import ColorMap from "../../../modules/ColorMap";
import { VariantOptions } from "../../../modules/testUtils";
import { SimpleCollapse } from "../SimpleCollapse";

describe("SimpleCollapse", ()=>{
    it("Should render as expected", ()=>{
        const {container} = render(<SimpleCollapse headerContent="test" />);
        expect(container).toMatchSnapshot();
    });

    describe("Variants", ()=>{
        VariantOptions.forEach(variant=>{
            it(`Respects ${variant} variant`, ()=>{
                const component = render(
                    <SimpleCollapse headerContent={`testing ${variant} variant`} variant={variant} />
                );
                const headerContainer = component.container.querySelector("div.collapse-title");
                const bodyContainer = component.container.querySelector("div.collapse-body") as HTMLElement;
                const colors = ColorMap.get(variant);

                expect(headerContainer).toHaveClass(colors.bg, colors.text);
                expect(bodyContainer).toHaveClass(colors.bg, colors.text);

                expect(headerContainer).toMatchSnapshot();
                expect(bodyContainer).toMatchSnapshot();
            });
        });
    });

    it("Respects the LeftIcon prop", ()=>{
        const {container} = render(
            <SimpleCollapse
                LeftIcon={<FontAwesomeIcon icon={faGithub} className="my-icon h-5 w-5" />}
                headerContent="Testing"
            >
                <p>Hey everyone</p>
            </SimpleCollapse>
        );

        expect(container.querySelector("svg.my-icon")).toBeInTheDocument();
    });

    it("Respects the className prop", ()=>{
        const {container} = render(
            <SimpleCollapse
                headerContent="Testing"
                className="find-me"
            >
                <p>Hey everyone</p>
            </SimpleCollapse>
        );

        expect(container.querySelector("div.find-me")).toBeInTheDocument();
    });

    it("Respects the rounded prop on collapsed state", ()=>{
        const {container} = render(
            <SimpleCollapse
                headerContent="Testing"
                rounded={true}
            >
                <p>Hey everyone</p>
            </SimpleCollapse>
        );
        expect(container.querySelector("div.collapse-title")).toHaveClass("rounded-md", "transition-all", "duration-[1s]");
        expect(container.querySelector("div.collapse-body")).toHaveClass("rounded-b-md");
    });

    it("Respects the rounded prop on expanded state", ()=>{
        const {container} = render(
            <SimpleCollapse
                headerContent="Testing"
                rounded={true}
                defaultExpand
            >
                <p>Hey everyone</p>
            </SimpleCollapse>
        );
        expect(container.querySelector("div.collapse-title")).toHaveClass("rounded-t-md");
        expect(container.querySelector("div.collapse-body")).toHaveClass("rounded-b-md");
    });
});
