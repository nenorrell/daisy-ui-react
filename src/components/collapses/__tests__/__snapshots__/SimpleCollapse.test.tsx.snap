// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCollapse Should render as expected 1`] = `
<div>
  <div
    class="collapse"
    tabindex="0"
  >
    <div
      class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-neutral text-neutral-content rounded-md transition-all duration-[1s]"
    >
      <p
        class="w-full text-lg"
      >
        test
      </p>
      <div
        class="items-end transition ease-in-out -rotate-90"
      >
        <svg
          aria-hidden="true"
          class="w-4 h-4"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-neutral text-neutral-content"
    />
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects accent variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-accent text-accent-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing accent variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects accent variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-accent text-accent-content"
/>
`;

exports[`SimpleCollapse Variants Respects base-100 variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-base-100 text-base-100-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing base-100 variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects base-100 variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-base-100 text-base-100-content"
/>
`;

exports[`SimpleCollapse Variants Respects base-200 variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-base-200 text-base-200-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing base-200 variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects base-200 variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-base-200 text-base-200-content"
/>
`;

exports[`SimpleCollapse Variants Respects base-300 variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-base-300 text-base-300-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing base-300 variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects base-300 variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-base-300 text-base-300-content"
/>
`;

exports[`SimpleCollapse Variants Respects error variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-error text-error-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing error variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects error variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-error text-error-content"
/>
`;

exports[`SimpleCollapse Variants Respects error variant 3`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-error text-error-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing error variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects error variant 4`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-error text-error-content"
/>
`;

exports[`SimpleCollapse Variants Respects info variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-info text-info-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing info variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects info variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-info text-info-content"
/>
`;

exports[`SimpleCollapse Variants Respects info variant 3`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-info text-info-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing info variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects info variant 4`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-info text-info-content"
/>
`;

exports[`SimpleCollapse Variants Respects neutral variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-neutral text-neutral-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing neutral variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects neutral variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-neutral text-neutral-content"
/>
`;

exports[`SimpleCollapse Variants Respects primary variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-primary text-primary-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing primary variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects primary variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-primary text-primary-content"
/>
`;

exports[`SimpleCollapse Variants Respects secondary variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-secondary text-secondary-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing secondary variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects secondary variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-secondary text-secondary-content"
/>
`;

exports[`SimpleCollapse Variants Respects success variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-success text-success-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing success variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects success variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-success text-success-content"
/>
`;

exports[`SimpleCollapse Variants Respects success variant 3`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-success text-success-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing success variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects success variant 4`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-success text-success-content"
/>
`;

exports[`SimpleCollapse Variants Respects warning variant 1`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-warning text-warning-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing warning variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects warning variant 2`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-warning text-warning-content"
/>
`;

exports[`SimpleCollapse Variants Respects warning variant 3`] = `
<div
  class="collapse-title cursor-pointer flex items-center justify-center px-4 bg-warning text-warning-content rounded-md transition-all duration-[1s]"
>
  <p
    class="w-full text-lg"
  >
    testing warning variant
  </p>
  <div
    class="items-end transition ease-in-out -rotate-90"
  >
    <svg
      aria-hidden="true"
      class="w-4 h-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`SimpleCollapse Variants Respects warning variant 4`] = `
<div
  class="collapse-body ease-linear duration-[.25s] transition-[height] overflow-hidden h-0 bg-warning text-warning-content"
/>
`;
